system-test/system.ts

```diff
@@ -14,4 +14,140 @@
  * limitations under the License.
  */
 
-console.warn(`no system tests available 👻`);
+import * as cp from 'child_process';
+import * as fs from 'fs';
+import * as path from 'path';
+import * as rimraf from 'rimraf';
+import * as util from 'util';
+
+const mkdir = util.promisify(fs.mkdir);
+const readFile = util.promisify(fs.readFile);
+const writeFile = util.promisify(fs.writeFile);
+const rmrf = util.promisify(rimraf);
+
+const baseRepoUrl = 'https://github.com/googleapis/';
+const gaxPath = process.cwd();
+const testDir = path.join(process.cwd(), 'system-test-run');
+
+interface PackageJson {
+  dependencies: {[name: string]: string;};
+}
+
+interface ExecuteResult {
+  stdout: string;
+  stderr: string;
+}
+
+async function execute(command: string, cwd?: string): Promise<ExecuteResult> {
+  cwd = cwd || process.cwd();
+  const maxBuffer = 10 * 1024 * 1024;
+  console.log(`Execute: ${command} [cwd: ${cwd}]`);
+  return new Promise<ExecuteResult>((resolve, reject) => {
+    cp.exec(command, {cwd, maxBuffer}, (err, stdout, stderr) => {
```

Comment:
So it kinda depends on what you're looking for here, but I usually prefer to do `cp.spawn` using `stdio: inherit`.  If you do it that way, the output is written directly to stderr as the program executes, and it does require the promise wrapping like you're doing here.  If you explicitly want `cp.exec`, then you don't need to do the promise wrapping really - you could just use `util.promisify`.

Reply:
In this place I need to grab its `stdout` to parse it later, so I'd better leave it as is. For `spawn` below, I'll do `inherit` indeed, that will save some lines.